freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -167,3 +180,156 @@ def _filter_images_with_higher_rpm_nvr(self, event, rebuild_images_list):
+        :rtype: set
+        """
+        module_name_stream_version_set = set()
+        with koji_service(
+            conf.koji_profile, log, login=False, dry_run=self.dry_run
+        ) as session:
+            for advisory_module_nvr in self.advisory_module_nvrs:
+                mmd = session.get_modulemd(advisory_module_nvr)
+                name = mmd.get_module_name()
+                stream = mmd.get_stream_name()
+                version = mmd.get_version()
+                module_name_stream_version_set.add(
+                    ":".join([name, stream, str(version)])
+                )
+
+        return module_name_stream_version_set
+
+    def _prepare_data_for_compose(
+        self, original_odcs_compose_ids, module_name_stream_version_set
+    ):
+        """
+        Prepare data before recording compose data in build or
+        creating a new compose and linking it to a build.
+
+        :param original_odcs_compose_ids list: A list of compose ids
+            in a image.
+        :param module_name_stream_version_set set: A set of module's metadata
+            in format NAME:STREAM:VERSION of an advisory.
+        :return: a tuple of compose data. The first item is a set of
+            compose ids which will be renewed in build, the second item
+            is a set of required data for creating new composes
+        :rtype: tuple
+        """
+        renewed_odcs_compose_set = set()
+        new_odcs_compose_set = set()
+        module_name_stream_set = {v.rsplit(":", 1)[0] for v in module_name_stream_version_set}
+        for compose_id in original_odcs_compose_ids:
+            compose = self.odcs.get_compose(compose_id)
+            source_type = compose.get("source_type")
+            if source_type != PungiSourceType.MODULE:
+                renewed_odcs_compose_set.add(compose_id)
+                continue
+
+            name_stream_set = set()
+            mapping = dict()
+            source_value = compose.get("source", "")
+            for source in source_value.split():
+                source = source.strip()
+                name_stream = source.rsplit(":", 2)[0]
+                name_stream_version = source.rsplit(":", 1)[0]
+                name_stream_set.add(name_stream)
+                mapping[name_stream] = name_stream_version
+
+            # If all the sources are not in advisory, renew them
+            inter_section = name_stream_set.intersection(module_name_stream_set)
+            if not inter_section:
+                renewed_odcs_compose_set.add(compose_id)
+            else:
+                # If some module in advisory, create composes for all differences,
+                # inter_section part will be added in following step
+                difference = name_stream_set.difference(module_name_stream_set)
+                for item in difference:
+                    new_odcs_compose_set.add(mapping[item])
+            # All modules in advisory should create compose
```

Comment:
Why all modules in advisory should be enabled for all impacted images? If there are multiple modules in advisory, each module can impact different images, for example, an advisory has `module-a` and `module-b`, `module-a` only impacts `a-flatpak-image`, `module-b` only impacts `b-flatpak-image`, in this case, you only need to request `module-a` compose for image `a-flatpak-image`.

Reply:
Ack. Fixed